hnn_core/drives.py

```diff
@@ -229,35 +224,33 @@ def _drive_cell_event_times(drive_type, drive_conn, dynamics,
     elif len(matches) > 1:
         raise ValueError('Ambiguous external drive: %s' % drive_type)
 
-    # Return values not checked: False if all weights for given drive type
-    # are zero. Designed to be silent so that zeroing input weights
-    # effectively disables each.
-    n_ampa_nmda_weights = (len(drive_conn['ampa'].keys()) +
-                           len(drive_conn['nmda'].keys()))
-    target_syn_weights_zero = True if n_ampa_nmda_weights == 0 else False
-
     event_times = list()
-    if drive_type == 'poisson' and not target_syn_weights_zero:
-        event_times = _create_extpois(
-            t0=dynamics['tstart'],
-            T=dynamics['tstop'],
-            lamtha=dynamics['rate_constant'][drive_conn['target_type']],
-            prng=prng)
-    elif not target_syn_weights_zero and (drive_type == 'evoked' or
-                                          drive_type == 'gaussian'):
+    if drive_type == 'poisson':
+        # Poisson rate constant in Network.add_poisson_drive() can be
+        # target specific or not
```

Comment:
this has been on my mind for a while. I'd like to improve the commenting practices in HNN-core. We have lots of superfluous comments that are sometimes distracting. See: 

https://stackoverflow.blog/2021/07/05/best-practices-for-writing-code-comments/

"Rule 1: Comments should not duplicate code"